client/modules/driver/reducers/helpers.js

```diff
@@ -8,15 +8,8 @@ export const getGoogleRoute = (waypoints, optimize) => {
     location: {lat: point.lat, lng: point.lng}
   }))
 
-  let resolve, reject
-  const result = new Promise((res, rej) => {
-    resolve = res
-    reject = rej
-  })
-
-  let directionsService
-  try {
-    directionsService = new google.maps.DirectionsService
+  return new Promise((resolve, reject) => {
```

Comment:
Yeah that's much cleaner. Could also do:
```js
return Promise.resolve().then(() => {
  if (ok) {
    return result  // instead of resolve(result)
  } else {
    throw error  // instead of reject(error)
  }
})
```